proposal/color-4-new-spaces-js.d.ts.md

```diff
@@ -127,61 +127,107 @@ get space(): KnownColorSpace;
 
 #### `channels`
 
-Returns an array of channel values for [`internal`], with missing channels
-converted to `0`.
+Returns an array of channel values (excluding `alpha`) for [`internal`], with
+[missing channels][missing components] converted to `0`.
+
+[missing components]: ./color-4-new-spaces.md#missing-components
 
 ```ts
-get channels(): string | number;
+get channels(): [number, number, number];
 ```
 
 #### `channelsOrNull`
 
-Returns an array of channel values for [`internal`], with missing channels
-converted to `null`.
+Returns an array of channel values (excluding `alpha`) for [`internal`], with
+[missing channels][missing components] converted to `null`.
 
 ```ts
-get channelsOrNull(): ChannelValue;
+get channelsOrNull(): [ChannelValue, ChannelValue, ChannelValue];
 ```
 
 #### `isLegacy`
 
-Returns whether [`internal`] is in a legacy color space (`rgb`, `hsl`, or
+Returns whether [`internal`] is in a [legacy color space] (`rgb`, `hsl`, or
 `hwb`).
 
 ```ts
 get isLegacy(): boolean;
 ```
 
+[legacy color space]: ./color-4-new-spaces.md#legacy-color
+
 #### `isInGamut`
 
-Returns whether [`internal`] is in-gamut for its color space (as opposed to
+Returns whether [`internal`] is [in-gamut] for its color space (as opposed to
 having one or more of its channels out of bounds, like `rgb(300 0 0)`).
 
 ```ts
 get isInGamut(): boolean;
 ```
 
+[in-gamut]: ./color-4-new-spaces.md#coloris-in-gamut
+
 #### `channel`
 
-Returns the value of the given `channel` in [`internal`], after converting it to
-`space` if necessary. It should be used instead of the old channel-specific
-functions such as `color.red()` and `color.hue()`.
+Returns the value of the given `channel` in [`internal`]. If `space` is set, it
+will first [convert] to that `space`.
 
 ```ts
-channel(options: {
-  channel: ChannelName;
-  space?: KnownColorSpace;
-}): ChannelValue;
+channel(channel: ChannelName): ChannelValue;
+channel(
+  channel: ChannelNameHSL,
+  options?: {space: ColorSpaceHSL}
```

Comment:
I think these should be mandatory, because what we're trying to express in these types is (for example) _if_ you pass an HSL color space _then_ you have to pass an HSL channel name. If the color space isn't passed, it'll just use the first overload.